notebook2/notebook/notebook.py

```diff
@@ -473,26 +420,28 @@ def wait_websocket(ws):
     if notebook is None:
         return
 
-    pod_name = notebook['pod_name']
-    svc_name = notebook['svc_name']
+    pod_uuid = notebook['pod_uuid']
+
+    url = f'{external_url_for("")}instance-ready/{pod_uuid}/'
+
     while True:
         try:
-            response = requests.head(f'https://notebook2.hail.is/instance-ready/{svc_name}/',
-                                     timeout=1)
-            if response.status_code < 500:
-                log.info(f'HEAD on jupyter succeeded for {svc_name} {pod_name} response: {response}')
+            response = requests.head(url, timeout=1)
+            if response.status_code == 405 or response.status_code == 302:
+                log.info(f'HEAD on jupyter succeeded for pod_uuid: {pod_uuid} : response: {response}')
                 # if someone responds with a 2xx, 3xx, or 4xx, the notebook
                 # server is alive and functioning properly (in particular, our
                 # HEAD request will return 405 METHOD NOT ALLOWED)
+                # or 302 if the user had previously logged in (and has nec. cookie)
```

Comment:
I do not understand this. The notebook leader is making the `HEAD` request. How does the notebook leader have the user's cookie? We should only ever receive `405 METHOD NOT ALLOWED` from a `HEAD /` to Jupyter because Jupyter does not (AFAICT) support the `HEAD` verb on it's root.

Reply:
If you issue this request after the user has logged in, it will return a 302. I don’t know how that is possible without persisting state.

Reply:
@danking solved. 

There is a comment about potential /wait requests after 405, due to Kubernetes container state having a race condition with pod reachability. I've uploaded an example and included in the comment: https://youtu.be/dp-RYuH4tmw

Reply:
Also,`/wait` is substantially changing (in the fine grained permissions pr), since its foremost task is soon to be watching the kubernetes cluster, using the user/notebook-appropriate selector, for state updates.